src/components/StakingVaults/hooks/useFetchOpportunities.tsx

```diff
@@ -0,0 +1,98 @@
+import type { AccountId, AssetId } from '@shapeshiftoss/caip'
+import {
+  avalancheChainId,
+  bchChainId,
+  bscChainId,
+  btcChainId,
+  cosmosChainId,
+  dogeChainId,
+  fromAccountId,
+  ltcChainId,
+} from '@shapeshiftoss/caip'
+import { useQuery } from '@tanstack/react-query'
+import { useMemo } from 'react'
+import { useSelector } from 'react-redux'
+import { useFeatureFlag } from 'hooks/useFeatureFlag/useFeatureFlag'
+import {
+  useGetZapperAppsBalancesOutputQuery,
+  useGetZapperUniV2PoolAssetIdsQuery,
+} from 'state/apis/zapper/zapperApi'
+import {
+  fetchAllOpportunitiesIdsByChainId,
+  fetchAllOpportunitiesMetadataByChainId,
+  fetchAllOpportunitiesUserDataByAccountId,
+} from 'state/slices/opportunitiesSlice/thunks'
+import type { PortfolioAccount } from 'state/slices/portfolioSlice/portfolioSliceCommon'
+import {
+  selectPortfolioAccounts,
+  selectPortfolioAssetIds,
+  selectPortfolioLoadingStatus,
+  selectWalletAccountIds,
+} from 'state/slices/selectors'
+
+export const useFetchOpportunities = () => {
+  const portfolioLoadingStatus = useSelector(selectPortfolioLoadingStatus)
+  const requestedAccountIds = useSelector(selectWalletAccountIds)
+  const portfolioAssetIds = useSelector(selectPortfolioAssetIds)
+  const portfolioAccounts = useSelector(selectPortfolioAccounts)
+  const DynamicLpAssets = useFeatureFlag('DynamicLpAssets')
+
+  const { isLoading: isZapperAppsBalancesOutputLoading } = useGetZapperAppsBalancesOutputQuery()
+  const { isLoading: isZapperUniV2PoolAssetIdsLoading } = useGetZapperUniV2PoolAssetIdsQuery(
+    undefined,
+    { skip: !DynamicLpAssets },
+  )
+
+  const queryKey: [
+    string,
+    {
+      requestedAccountIds: AccountId[]
+      portfolioAssetIds: AssetId[]
+      portfolioAccounts: Record<AccountId, PortfolioAccount>
+    },
+  ] = useMemo(() => {
+    return ['fetchAllOpportunities', { requestedAccountIds, portfolioAssetIds, portfolioAccounts }]
+  }, [portfolioAccounts, portfolioAssetIds, requestedAccountIds])
+
+  const { isLoading } = useQuery({
```

Comment:
@woodenfurniture queries go stale immediately, and have a default garbage collect time of 5mn, do we want to `Infinity` both of these here?

Reply:
Absolutely we do, nice catch